locals.tf

```diff
@@ -84,8 +84,12 @@ locals {
   service_account_amp_ingest_name = format("%s-%s", module.aws_eks.cluster_id, "amp-ingest")
   service_account_amp_query_name  = format("%s-%s", module.aws_eks.cluster_id, "amp-query")
 
-  # Configuration for managing add-ons via GitOps.
-  gitops_add_on_config = {
-    awsForFluentBit = var.aws_for_fluentbit_enable ? module.aws_for_fluent_bit[0].gitops_config : null
+  # Configuration for managing add-ons via ArgoCD.
+  argocd_add_on_config = {
+    agones                    = var.agones_enable ? module.agones[0].argocd_gitops_config : null
+    awsForFluentBit           = var.aws_for_fluentbit_enable ? module.aws_for_fluent_bit[0].argocd_gitops_config : null
+    awsLoadBalancerController = var.aws_lb_ingress_controller_enable ? module.aws_load_balancer_controller[0].argocd_gitops_config : null
+    certManager               = var.cert_manager_enable ? module.cert_manager[0].argocd_gitops_config : null
+    clusterAutoscaler         = var.cluster_autoscaler_enable ? module.cluster_autoscaler[0].argocd_gitops_config : null
```

Comment:
I think we also need to another condition here  `var.argocd_manage_add_ons`

e.g., `agones                    = var.agones_enable && var.argocd_manage_add_ons ? module.agones[0].argocd_gitops_config : null`

Reply:
If `argocd_manage_add_ons` is set to `false` (default), this value is not used...so probably not needed?